### PR TITLE
Implemented a basic API for querying attendees

### PR DESCRIPTION
**Pull request by EliAndrewC**
Implements the API requested at #1163 over JSON-RPC.  Two methods have been implemented:
- ``attendee.lookup(badge_num)``: returns a single attendee with the given badge number as a dictionary.  If the attendee doesn't exist, a dictionary with an ``error`` key will be returned.
- ``attendee.search(query)``: returns an array of all attendees which match the search string, using the same search logic implemented on the admin Attendees list.  An empty array will be returned if no attendees match the query.

Note: this will do absolutely nothing at all until @binary1230 adds some Nginx config, because Sideboard exposes a ``/jsonrpc`` web handler, but Nginx is currently configured to only proxy requests under ``/uber``.

However, if you were running locally on the box, where you can connect directly to Sideboard on ``localhost:8282``, then you could run the following script to demonstrate the new JSON-RPC service:

```python
from pprint import pprint
from rpctools.jsonrpc import ServerProxy

service = ServerProxy('http://localhost:8282/jsonrpc')
pprint(service.attendee.lookup(badge_num=1))
pprint(service.attendee.search('Test'))
```

The above code uses the https://github.com/appliedsec/rpctools#rpctools Python library as a JSON-RPC client, but any standards-compliant JSON-RPC client in any language would do the same.  Here's an example of what the above prints out on my box (note that the output is just the same "Test Developer" which is returned by both queries:

```python
{'_model': 'Attendee',
 'assigned_depts_labels': ['Arcade'],
 'badge_status_label': 'New',
 'badge_type_label': 'Staff',
 'cellphone': '',
 'checked_in': None,
 'ec_phone': '',
 'email': 'magfest@example.com',
 'first_name': 'Test',
 'food_restrictions': None,
 'full_name': 'Test Developer',
 'id': '4b793f7b-8e91-4549-8739-4da07fd4d9f4',
 'is_dept_head': False,
 'last_name': 'Developer',
 'ribbon_label': 'no ribbon',
 'shifts': [{'_model': 'Shift',
             'id': '1b2f2bfc-a316-4ce1-81aa-d607edb57a2a',
             'job': {'_model': 'Job',
                     'description': '',
                     'end_time': '2015-09-11 14:00:00.000000',
                     'extra15': False,
                     'id': 'ee139641-6ca4-48a6-8be2-b7d7c59c365b',
                     'location_label': 'Arcade',
                     'name': 'Arcade Manager',
                     'start_time': '2015-09-11 12:00:00.000000',
                     'type_label': 'Regular'},
             'worked_label': 'SELECT A STATUS'}],
 'staffing': True,
 'weighted_hours': 2.0,
 'worked_hours': 0.0,
 'zip_code': ''}
[{'_model': 'Attendee',
  'assigned_depts_labels': ['Arcade'],
  'badge_status_label': 'New',
  'badge_type_label': 'Staff',
  'cellphone': '',
  'checked_in': None,
  'ec_phone': '',
  'email': 'magfest@example.com',
  'first_name': 'Test',
  'food_restrictions': None,
  'full_name': 'Test Developer',
  'id': '4b793f7b-8e91-4549-8739-4da07fd4d9f4',
  'is_dept_head': False,
  'last_name': 'Developer',
  'ribbon_label': 'no ribbon',
  'shifts': [{'_model': 'Shift',
              'id': '1b2f2bfc-a316-4ce1-81aa-d607edb57a2a',
              'job': {'_model': 'Job',
                      'description': '',
                      'end_time': '2015-09-11 14:00:00.000000',
                      'extra15': False,
                      'id': 'ee139641-6ca4-48a6-8be2-b7d7c59c365b',
                      'location_label': 'Arcade',
                      'name': 'Arcade Manager',
                      'start_time': '2015-09-11 12:00:00.000000',
                      'type_label': 'Regular'},
              'worked_label': 'SELECT A STATUS'}],
  'staffing': True,
  'weighted_hours': 2.0,
  'worked_hours': 0.0,
  'zip_code': ''}]
```

I'll explain what I did inline, but mostly I'll just be referring to the Sideboard docs on this stuff.

### CODE REVIEW

File: uber/api.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    def lookup(self, badge_num):
+        with Session() as session:
+            attendee = session.query(Attendee).filter_by(badge_num=badge_num).first()
+            return attendee.to_dict(fields) if attendee else {'error': 'No attendee found with Badge #{}'.format(badge_num)}
+
+    def search(self, query):
+        with Session() as session:
+            return [a.to_dict(fields) for a in session.search(query).all()]
+
+services.register(AttendeeLookup(), 'attendee')
```
Comment (EliAndrewC) [inline uber/api.py]: See the Sideboard tutorial for an explanation of how services work:
https://github.com/magfest/sideboard/blob/master/docs/source/index.rst#service

The tl;dr version is that there's a ``sideboard.lib.services`` object, and you can register objects such as modules or class instances, and that will cause Sideboard to expose those as JSON-RPC and WebSocket RPC methods.

File: uber/api.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+from uber.common import *
+
```
Comment (EliAndrewC) [inline uber/api.py]: This is the format expected by the ``to_dict()`` method which Sideboard provides on anything which uses its ``@declarative_base`` decorator.  This is extensively documented at
https://github.com/magfest/sideboard/blob/master/sideboard/lib/sa/_crud.py#L1

The tl'dr version is that we generate a dictionary from the model's fields and can optionally tell it which fields to use, including which foreign key relationships it should follow.

**Comment by kitsuta** [comment]
+1 OMHBIT

**Comment by binary1230** [comment]
damn yo.  that is really really cool.  +1   will work on the nginx side

**Comment by EliAndrewC** [comment]
> +1 OMHBIT

The full acronym is "Over My Head But I Trust You", which would be OMHBITY.  I assume it was just a typo that the Y was omitted, but I like to imagine that it was deliberate.

"This is over my head, but I trust."
"You mean you trust me?"
"No, definitely not.  I just trust in general.  But not you."

:P

**Comment by kitsuta** [comment]
Haha, yes. Actually it turns out that the mobile website doesn't let you edit your comments, so even though I noticed the typo after I posted I still couldn't fix it!

**Pull request merged**